euphonic/cli/utils.py

```diff
@@ -1,13 +1,86 @@
-import os
-from typing import List, Tuple
 from argparse import ArgumentParser
+import json
+import os
+import pathlib
+from typing import Any, Dict, List, Tuple, Union
+
+import numpy as np
+from pint import UndefinedUnitError
+import seekpath
+
+from euphonic import ForceConstants, QpointPhononModes, Quantity, ureg
+Unit = ureg.Unit
+
+
+def force_constants_from_file(filename: Union[str, os.PathLike]
+                              ) -> ForceConstants:
+    """
+    Load force constants data from file
+
+    Parameters
+    ----------
+    filename
+        Data file
 
-from euphonic import QpointPhononModes
+    Returns
+    -------
+    ForceConstants
+    """
+    path = pathlib.Path(filename)
+    if path.suffix == '.yaml':
+        return ForceConstants.from_phonopy(
+            path=path.parent, summary_name=path.name)
+    elif path.suffix in ('.castep_bin', '.check'):
+        return ForceConstants.from_castep(filename)
+    elif path.suffix == '.json':
+        return ForceConstants.from_json_file(filename)
+    else:
+        raise ValueError("File not recognised. Should have extension "
+                         ".yaml (phonopy), .castep_bin or .check "
+                         "(castep) or .json (JSON from Euphonic).")
 
 
-def load_data_from_file(filename: str) -> QpointPhononModes:
+def modes_from_file(filename: Union[str, os.PathLike]
+                    ) -> QpointPhononModes:
     """
-    Load castep data from filename and return it
+    Load phonon mode data from file
+
+    Parameters
+    ----------
+    filename
+        Data file
+
+    Returns
+    -------
+    QpointPhononmodes
+    """
+    path = pathlib.Path(filename)
+    if path.suffix == '.phonon':
+        return QpointPhononModes.from_castep(path)
+    elif path.suffix == '.json':
+        return QpointPhononModes.from_json_file(path)
+    else:
+        raise ValueError("File not recognised. Should have extension "
+                         ".phonon (castep) or .json (JSON from Euphonic).")
+
+
+def _load_json(filename: Union[str, os.PathLike]
+               ) -> Union[QpointPhononModes, ForceConstants]:
+    with open(filename, 'r') as f:
+        data = json.load(f)
+
+    if 'force_constants' in data:
+        return force_constants_from_file(filename)
+    elif 'eigenvectors' in data:
+        return modes_from_file(filename)
+    else:
+        raise ValueError("Could not identify Euphonic data in JSON file.")
+
+
+def load_data_from_file(filename: Union[str, os.PathLike]
```

Comment:
I've just tried `intensity-map` with a force_constants.hdf5 file and it appears there's no code to handle that in `force_constants_from_file`

I think we should probably also allow reading from phonopy band.yaml/hdf5 qpoints.yaml/hdf5 etc. Annoyingly because of the variety of Phonopy files the best approach I can immediately think of is to try reading it as `QpointPhononModes` and see if it works, and if not fall back to `ForceConstants`...

Reply:
Am I right in thinking that we can't _actually_ obtain a  ForceConstants from `force_constants.hdf5` because it needs to also read a `phonopy.yaml` to get the structure/supercell? So a "read force_constants.hdf5" function needs to also go looking for a phonopy.yaml?

Reply:
Yes... we could assume if a user gives us a `force_constants.hdf5`, that they've also provided a `phonopy.yaml` in the same directory? And assume the same for `bands/qpoints.yaml/hdf5`, it's probably a fair middle ground? Being able to input all the `from_phonopy` arguments in at the cli has the potential for arguments to become messy

Reply:
Does that mean we can assume that the files have not been renamed?

Reply:
For now I think so? I think its enough functionality for what is supposed to be a quick CL tool, if there is need later we can always add more arguments. If people have big ecosystems of phonopy files with different names they can always write their own script...

Reply:
Done!